src/features/hotspots/details/HotspotDetails.tsx

```diff
@@ -234,13 +210,23 @@ const HotspotDetails = () => {
               id: `witness-${numDays}-${w.timestamp}`,
             })) || []
           }
-          loading={loadingWitnessSums}
+          loading={loading}
         />
         <HotspotDetailChart
           title={t('hotspot_details.challenge_title')}
-          percentage={78}
+          subTitle={t('hotspot_details.challenge_sub_title')}
+          number={challengeSum?.toFixed(0)}
+          change={challengeChange}
           color={colors.purpleMain}
-          data={data[2]}
+          data={
+            challengeSums?.map((w) => ({
```

Comment:
I'd recommend breaking this out into a `useMemo`

Reply:
Updated